### PR TITLE
OBPIH-4153 Corrected values ATP in returns and validation of returnSh…

### PR DESCRIPTION
**Pull request by MagdaToczek**
…impentItem in shipmentService

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy [hunk 1]
```diff
@@ -155,7 +155,7 @@ class ProductAvailabilityService {
             SELECT
                 pli.binLocation,
                 pli.inventoryItem,
-                sum(pli.quantity)
+                sum(pli.quantity)*(count(distinct pl.requisition) + count(distinct pl.order) / count(*))
```
Comment (pmuchowski) [inline grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:158]: Why you did not use `pli.requisitionItem` and `pli.orderItem`?
Also I think you should add brackets before dividing by the count(*)

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy (hunk 1 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy]: I checked that with `pli.requisitionItem` and `pli.orderItem` it works too, so I changed it.

File: grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy [hunk 2]
```diff
@@ -757,9 +757,9 @@ class ShipmentService {
 
             def quantityAvailableToReturn
             if (withBinLocation) {
-                quantityAvailableToReturn = productAvailabilityService.getQuantityNotPickedInBinLocation(shipmentItem.inventoryItem, shipmentItem.binLocation)
+                quantityAvailableToReturn = productAvailabilityService.getQuantityNotPickedInBinLocation(shipmentItem.inventoryItem, shipmentItem.binLocation) + shipmentItem.quantity
```
Comment (pmuchowski) [inline grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy:760]: I think you should use quantity picked, not quantity,
right now in the shipmentItem there is method to get picked quantity for the requisition based shipment,
you could add similar method for the order

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy (hunk 2 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy]: Done

**Review by pmuchowski — APPROVED** [review]

**Pull request merged**